sdks/python/container/boot.go

```diff
@@ -172,12 +186,12 @@ func main() {
 		}
 	}
 
-	workerPoolId := os.Getenv(workerPoolIdEnv)
 	if workerPoolId != "" {
-		multiProcessExactlyOnce(materializeArtifactsFunc, "beam.install.complete."+workerPoolId)
+		defer multiProcessExactlyOnce(materializeArtifactsFunc, "beam.install.complete." + workerPoolId)()
 	} else {
 		materializeArtifactsFunc()
 	}
+	defer os.RemoveAll(venvDir)
```

Comment:
Cleanup is deferred until all sibling processes have finished, but I don't know what that means for the entire job/pipeline life cycle. This could potentially blow up if multiple independent worker processes are running in parallel.

Reply:
On the other hand, the code has never been safe against that scenario. `multiProcessExactlyOnce()` uses a mutex for running `actionFunc()`, which is only safe for child processes that share the same process memory. Since the entry point is called via Popen, this never happens anyway. So I am at a total loss what `multiProcessExactlyOnce()` is suppose to do in the first place.